service/worker/migration/activities.go

```diff
@@ -555,6 +572,17 @@ func (a *activities) canSkipWorkflowExecution(
 		return true, reasonZombieWorkflow, nil
 	}
 
+	// source and target cluster handles workflow retention separately. For a workflow which is abort to be deleted,
+	// it may be already deleted on target cluster but still exist on source, which cause verification delay.
+	// Here, we skip workflow of which retention time is close to current time to continue the verification.
+	if closeTime := resp.GetDatabaseMutableState().GetExecutionInfo().GetCloseTime(); closeTime != nil && ns != nil && ns.Retention() > 0 {
+		deleteTime := closeTime.Add(ns.Retention())
```

Comment:
Retention can change and this is not accurate. I think in the latest release YX stored the retention time in the mutable state.